File: minimum-changes-to-make-alternating-binary-string.js
/**
 * @param {string} s
 * @return {number}
 */
var minOperations = function(s) {
  let startChar = "0";
  let operations = 0;

  for (let i = 0; i < s.length; i++) {
      if (s[i] === startChar) {
          operations++;
      }

      startChar = startChar === "1" ? "0" : "1";
  }

  return Math.min(operations, s.length - operations);
};

File: minimum-changes-to-make-alternating-binary-string.py
class Solution:
    def minOperations(self, s: str) -> int:
        start_char = "0"
        operations = 0
        for char in s:
            if char == start_char:
                operations += 1
            start_char = "0" if start_char == "1" else "1"
        return min(operations, len(s) - operations)
